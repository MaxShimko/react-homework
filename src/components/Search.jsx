import React from 'react';
import { Row, Col } from 'react-bootstrap';

import '../styles/Search.scss';
import Toggler from './Toggler.jsx';

class Search extends React.Component {
    constructor(props) {
        super(props);

        this.filter = {type: 0, string: ''};

        this.state = {
            searchString: ''
        };
    }

    onChangeType(id) {
        this.filter.type = id;
        this.props.onSearch(this.filter);
    }

    onSearch() {
        this.filter.string = this.state.searchString;
        this.props.onSearch(this.filter);
    }

    handleInputChange(event) {
        this.setState({searchString: event.target.value});
    }

    render() {
        return <React.Fragment>
            <Row>
                <Col xs={8}>
                    <input type="text" placeholder="Search" value={this.state.searchString} onChange={this.handleInputChange.bind(this)} />
                </Col>
                <Col xs={4}>
                    <button onClick={this.onSearch.bind(this)}>Search</button>
                </Col>
            </Row>
            <div className="delimiter-2"></div>
            <Row>
                <Col>
                    <Toggler title="Search by" tabs={['Title', 'Genre']} onChange={this.onChangeType.bind(this)} />
                </Col>
            </Row>
        </React.Fragment>;
    }
}

export default Search;
