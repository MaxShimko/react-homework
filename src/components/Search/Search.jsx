import React from 'react';
import { Row, Col } from 'react-bootstrap';

import './style.scss';
import Toggler from '../Toggler/Toggler.jsx';

class Search extends React.Component {
    constructor(props) {
        super(props);

        this.filter = {type: 0, string: ''};

        this.state = {
            searchString: ''
        };
    }

    onChangeType(id) {
        this.filter.type = id;
        this.props.onSearch(this.filter);
    }

    onSearch() {
        this.filter.string = this.state.searchString;
        this.props.onSearch(this.filter);
    }

    handleInputChange(event) {
        this.setState({searchString: event.target.value});
    }

    render() {
        return (
            <React.Fragment>
                <Row>
                    <Col xs={8}>
                        <input id="test_search_input" type="text" placeholder="Search" value={this.state.searchString} onChange={this.handleInputChange.bind(this)} />
                    </Col>
                    <Col xs={4}>
                        <button id="test_search_button" onClick={this.onSearch.bind(this)}>Search</button>
                    </Col>
                </Row>
                <div className="delimiter-2" />
                <Row>
                    <Col>
                        <Toggler id="test_search_toggler" title="Search by" tabs={['Title', 'Genre']} onChange={this.onChangeType.bind(this)} />
                    </Col>
                </Row>
            </React.Fragment>
        );
    }
}

export default Search;
