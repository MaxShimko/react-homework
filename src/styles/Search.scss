input {
  width: 100%;
  outline: none;
  border: none;
  border-radius: 4px;
  background: rgba(#232323, 0.6);
  color: #fff;
  padding: 10px 15px;
  box-shadow: 0 0 10px 0 rgba(#000, 0.5);
  backdrop-filter: blur(4px);
}

button {
  width: 100%;
  border: none;
  background: #F65261;
  color: #fff;
  border-radius: 4px;
  padding: 10px 15px;
  text-transform: uppercase;
  &:focus {
    outline: none;
  }
}
