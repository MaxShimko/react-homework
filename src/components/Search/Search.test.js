import React from 'react';
import { shallow, mount } from 'enzyme';
import Search from './Search';
import {describe, expect, it} from '@jest/globals';
import toJson from 'enzyme-to-json';

describe('<Search />', () => {
    it('renders without crashing', () => {
        const component = shallow(<Search />);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('generate right filter', () => {
        const onSearch = jest.fn()

        const component = mount(<Search onSearch={onSearch} />);

        // const val = component.find('#test_search_input').prop('value');
        // console.log(val);
        expect(component.find('#test_search_input').prop('value')).toEqual('');

        component.find('#test_search_input').simulate('change', { target: {value: 'match string'} });
        component.find('#test_search_button').simulate('click');
        expect(onSearch).toHaveBeenCalledWith({type: 0, string: 'match string'});
    });
});
