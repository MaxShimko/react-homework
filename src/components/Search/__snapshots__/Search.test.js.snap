// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders without crashing 1`] = `
<Fragment>
  <Row
    noGutters={false}
  >
    <Col
      xs={8}
    >
      <input
        id="test_search_input"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </Col>
    <Col
      xs={4}
    >
      <button
        id="test_search_button"
        onClick={[Function]}
      >
        Search
      </button>
    </Col>
  </Row>
  <div
    className="delimiter-2"
  />
  <Row
    noGutters={false}
  >
    <Col>
      <Toggler
        id="test_search_toggler"
        onChange={[Function]}
        tabs={
          Array [
            "Title",
            "Genre",
          ]
        }
        title="Search by"
      />
    </Col>
  </Row>
</Fragment>
`;
